### ISSUE TITLE
include odot with cnmat-externals in package manager

### ISSUE THREAD

**Issue by ramagottfried** [CNMAT/CNMAT-odot#348]
We are going to release odot in the Max Package Manager soon and need to figure out how to best tie things together.

I was thinking that it might make the most sense to have one download with the Externals and Odot together and the Music-and-Computing package as a separate item.  

The repositories would stay separated, and just be combined with a script for submission to the Package Manager server.

Having _three_ separate downloads in the package manager seems maybe like too many to me, but curious what other people think.

Also, analyzer~ has odot objects in its help patch, so there will be some error messages if odot is not included.

@kulpajj @EdmundCampion @equilet @maccallum

**Comment by ramagottfried** [comment]
after working on cleaning up the folder structure today, I'm torn about it -- maybe it's not so bad to have three separate packages in the package manager... or maybe the externals and odot could be elegantly combined,...

**Comment by EdmundCampion** [comment]
In supporting the goal to build a larger community of users, I think making the leap of combining into one larger package for all CNMAT software components is the correct way to proceed. 

Is it possible to have a combined package called "CNMAT-Externals+ODOT" or "CNMAT-Softwares"?  Then, inside this combined package folder to have two separate packages  "CNMAT-Externals", and "CNMAT-ODOT" and eventually "CNMAT-???". 

As for Music and Computing (what will become Composing and Coding) we need to revisit the overall package structure vis a vis the types of content.  On one hand, we have new and evolving course materials and on the other, we have practical tools (some new 158 and some new MMJ-Depot derived).  These are merging in the sense that Pedagogy for 158 is now also being understood as useful to composing and production in the way that Depot materials are meant to be useful to that end.  

Basically, we need to review our earlier meetings on this and consider that many things might need to shift around.  I believe the over-arching reason for doing any of this is to encourage and enable greater involvement from students (undergraduate and graduate) and larger community, and in that order.

**Comment by kulpajj** [comment]
It would be good to integrate my "odot data flow + management" module with
the help files somewhere.   My students and a couple summer course students
have been learning some tricks from this module.  Even if not in this
current presentation, which is very much my own style of teaching in a
colorful way, there are examples in there that are very valuable.  John,
Jeff and I had input into this rather large module this summer during the
odot summer course, and Ilya before that.  And before that...1.5 years of
lessons with Rama!

The error messages in the Max window need to be cleaned up (by me) so you
guys can better check this out, but I think it deserves the attention of
future users/students of odot.  The summer course materials that are all to
Jeff's credit could just be synthesized with this module for what doesn't
yet overlap.

**Comment by equilet** [comment]
It's honestly not that big of a deal to download separate packages.  This is what everyone else who distributes does AFAIK.  I'd prefer not to bundle them.

**Comment by equilet** [comment]
Also worth mentioning that it's a 1-click install in the package manager.  I feel that if we offered an all-in-one everything package it would annoy most of our users outside of Berkeley, and then the question we need to be asking ourselves is "why did we go for the package manager in the first place?"

**Comment by equilet** [comment]
@kulpajj thanks for that input, but it is not germaine to this issue.  If you would like to be sure that it happens, please either open a new issue or see this one:  https://github.com/CNMAT/CNMAT-odot/issues/337

**Comment by equilet** [comment]
We can circumvent the issue of having everything available as a single download by simply packaging up modern releases from Github and offering them as a single download on the website that we update each semester.
